pkg/core/blockchain.go

```diff
@@ -61,21 +62,28 @@ const (
 	defaultStateSyncInterval   = 40000
 )
 
-// stateJumpStage denotes the stage of state jump process.
-type stateJumpStage byte
+type (
+	// stateChangeType is the type of the state modification process.
+	stateChangeType byte
+	// stateChangeStage denotes the stage of state modification process.
+	stateChangeStage byte
+)
 
+// Various state modification types.
 const (
-	// none means that no state jump process was initiated yet.
-	none stateJumpStage = 1 << iota
-	// stateJumpStarted means that state jump was just initiated, but outdated storage items
-	// were not yet removed.
-	stateJumpStarted
-	// newStorageItemsAdded means that contract storage items are up-to-date with the current
-	// state.
-	newStorageItemsAdded
-	// genesisStateRemoved means that state corresponding to the genesis block was removed
-	// from the storage.
-	genesisStateRemoved
+	stateJump stateChangeType = 1 << iota
+	stateReset
+)
+
+// A set of stages used to split state jump / state reset into atomic operations.
+const (
+	// none means that no state jump or state reset process was initiated yet.
+	none stateChangeStage = 1 << iota
+	stateChangeStage1
+	stateChangeStage2
+	stateChangeStage3
+	stateChangeStage4
+	stateChangeStage5
```

Comment:
These are very cryptic compared to the old ones. Can we just extend the list with the flags needed?

Reply:
Yes, we even have one bit left for future usage after all changes.